pyvista/plotting/camera.py

```diff
@@ -618,3 +644,70 @@ def copy(self):
             setattr(new_camera, attr, value)
 
         return new_camera
+
+    def tight(self, padding=0, adjust_render_window=True):
+        """Adjust the camera position so that the actors fill the entire renderer.
+
+        Parameters
+        ----------
+        padding : int, optional
+            Additional padding around the actors.
+
+        adjust_render_window : bool, optional
+            Adjust the size of the render window as to match the dimensions of
+            the visible actors.
+
+        Notes
+        -----
+        This only works with 2D images exposed in the XY plane.
+
+        Examples
+        --------
+        >>> import pyvista as pv
+        >>> from pyvista import examples
+        >>> puppy = examples.download_puppy()
+        >>> pl = pv.Plotter(border=True, border_width=5)
+        >>> _ = pl.add_mesh(puppy, rgb=True)
+        >>> pl.camera.tight(adjust_render_window=True)
+        >>> pl.show()
+
+        """
+        # inspired by vedo resetCamera. Thanks @marcomusy!
+        x0, x1, y0, y1, z0, z1 = self._renderer.ComputeVisiblePropBounds()
+        if z1 - z0 > 1e-5:
+            warnings.warn(
+                '`tight` should only be used with actors that lie only on the ' 'XY plane'
+            )
+
+        self.enable_parallel_projection()
+
+        self._renderer.ComputeAspect()
+        aspect = self._renderer.GetAspect()
+        angle = np.pi * self.view_angle / 180.0
+        dx, dy = (x1 - x0) * 0.999, (y1 - y0) * 0.999
```

Comment:
Does the factor 0.999 makes this _slightly_ zoomed in?  If so, I think it would be better to adjust the default padding instead.  This way it is very explicit to the user what is going on.  matplotlib `tight_layout` has some minor amount of extra padding in my experience.

Reply:
I've removed this minor zoom. As for the padding, I think it's better to have zero padding by default given that our background is generally not the same color as the image. Plus, default behavior for "tight" should be "tight" and not "kinda tight".

Fixed in 91464f77d4791d8395b51c1de8299f05a4694fa8.